File: LeetCode/1590/golang/lc1590_test.go
package lc1590

import "testing"

func Test_minSubarray(t *testing.T) {
	testcases := []struct {
		nums     []int
		p        int
		expected int
	}{
		{[]int{3, 1, 4, 2}, 6, 1},
		{[]int{6, 3, 5, 2}, 9, 2},
		{[]int{1, 2, 3}, 3, 0},
		{[]int{1, 2, 3}, 7, -1},
	}

	for i, tc := range testcases {
		actual := minSubarray(tc.nums, tc.p)
		if actual != tc.expected {
			t.Errorf("Testcase %02d (%v | %d) failed: want %d, got %d",
				i+1, tc.nums, tc.p, tc.expected, actual)
		}
	}
}

File: LeetCode/1590/python/lc1590_test.py
import unittest

from lc1590 import Solution

class Test_minSubarray(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.minSubarray([3, 1, 4, 2], 6), 1) # remove [4], [3, 1, 2] remains.
        self.assertEqual(s.minSubarray([6, 3, 5, 2], 9), 2) # remove [5, 2], [6, 3] remains.
        self.assertEqual(s.minSubarray([1, 2, 3], 3), 0)    # remove [], [1, 2, 3] remains.
        self.assertEqual(s.minSubarray([1, 2, 3, 4], 11), -1) # no way to remove a subarray.


if __name__ == "__main__":
    unittest.main()
